src/features/vault/redux/fetchBalances.js

```diff
@@ -54,8 +54,11 @@ export function fetchBalances({ address, web3, tokens }) {
 
           balanceResults.forEach(balanceResult => {
             newTokens[balanceResult.symbol] = {
-              ...tokens[balanceResult.symbol],
-              tokenBalance: balanceResult.balance,
+              ...newTokens[balanceResult.symbol],
+              tokenBalance: convertBalanceToDecimals(
+                balanceResult.balance,
+                tokens[balanceResult.symbol].decimals
+              ),
```

Comment:
Please preserve `tokenBalance` value as it comes from the source. Display data should be adjusted in the view, not in the state.